### PR TITLE
Submission: Atris Labs v8 (audited seed42, clean branch)

### PR DESCRIPTION
**Pull request by keshav55**
This PR supersedes #515 with a clean submission-only branch.

Included files:
- `records/track_10min_16mb/2026-03-22_AtrisLabs_v8/README.md`
- `records/track_10min_16mb/2026-03-22_AtrisLabs_v8/submission.json`
- `records/track_10min_16mb/2026-03-22_AtrisLabs_v8/train.log`
- `records/track_10min_16mb/2026-03-22_AtrisLabs_v8/train_gpt.py`

Audited run included in this branch:
- `seed=42`
- `val_bpb=1.18069496`
- `val_loss=1.99355398`
- `bytes_total=14461499`
- `steps=6428` in `600.039s`

Why this replacement:
- keeps the diff to the submission folder only
- includes a real train log
- aligns README/JSON with the audited run in the branch
- avoids the earlier commit-history/co-author concerns on the submission branch

If preferred, I can close #515 after this replacement is reviewed.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Submission: Atris Labs v8 (audited seed42, clean branch)

**Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache


---

## PR #671 — Atris Labs v8 (audited seed42, clean branch)

**Head SHA:** c586e4149bb2b36c8079f97095a4bd8b6292913a
**Reported BPB:** 1.18069496
**File audited:** `records/track_10min_16mb/2026-03-22_AtrisLabs_v8/train_gpt.py`

---

### Check 1 — N-gram family bug (CLOSE trigger: target token in hash key)

**CLEAN.** The `BigramHash.bigram_hash()` method computes:

```python
out[..., 1:] = torch.bitwise_xor(36313 * t[..., 1:], 27191 * t[..., :-1]) % mod
```

where `t = input_ids` (the shifted-left context sequence, i.e., `tokens[:-1]`). At position `i`, the hash uses `input_ids[i]` (current context token) and `input_ids[i-1]` (previous context token). The target token at position `i` is `target_ids[i] = input_ids[i+1]`, which is absent from the hash. This is identical in structure to legal BigramHash (per PR #1413 reference). No oracle lookahead.

### Check 2 — Pre-Quant TTT (CLOSE trigger: multi-epoch AdamW on val_tokens without score-first)

**CLEAN.** No test-time training of any kind is present. The training loop runs exclusively on `train_loader.next_batch()` (FineWeb train shards). `val_tokens` is consumed only by `eval_val()`, which runs entirely under `torch.inference_mode()` with no gradient computation, no optimizer steps, and no backward passes. The final post-training `eval_val` call after dequantization is also inference-only.

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). Compliance audit via LLM agent (Sonnet) reviewing full train_gpt.py source. If this review misread your code, please call it out so I can re-audit manually.*
